bundle/libraries/switch_to_patched_wheels.go

```diff
@@ -0,0 +1,102 @@
+package libraries
+
+import (
+	"context"
+	"path/filepath"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/config"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/log"
+)
+
+type switchToPatchedWheels struct{}
+
+func (c switchToPatchedWheels) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
```

Comment:
While it is still possible to modify typed representation directly, we normally use `b.Config.Mutate` if we need to change the configuration. Any reason why you need to modify typed configuration directly

Reply:
Can you elaborate, why do you normally use b.Config.Mutate instead of typed configuration?

I prefer typed configuration because it's easier to read, to write and type-safe. There are a number of mutators doing that, so we know it works. 

I see in some cases when you change the structure you need to ensure that new elements have location of the source (e.g. in expand globs) so you have to use dyn values if you want locations to work. But that's not the case here, the structure is unchanged.

Reply:
Overall direction we agreed on back when libs/dyn was added is to use:
1. `b.Config.Mutate` for any config mutation
2. Typed configuration for any read only operations

The reason is that using only dynamic mutation will allow us to get rid of back and forth conversion on each mutator exit / entry, we would only need to convert dynamic representation to typed one. Dynamic representation will become a source of truth for the config. @pietern any additional context from you on this?

Reply:
I propose that we discuss this separately. There are some ideas on how to merge dyn.Value and typed configuration to get benefits of both but the end result would look like neither and would require migration.

Either way, we have 16 mutators in initialize phase alone that make typed updates, so we'd need a migration plan.

Reply:
Discussed offline: it was confusion on my side, we don't force usage of Config.Mutate and need to use it only when needed